docs/topics/jvm/java-interop.md

```diff
@@ -124,33 +124,39 @@ Kotlin types. The compiler supports several flavors of nullability annotations,
 
 You can find the full list in the [Kotlin compiler source code](https://github.com/JetBrains/kotlin/blob/master/core/compiler.common.jvm/src/org/jetbrains/kotlin/load/java/JvmAnnotationNames.kt).
 
-### Annotating type parameters
+### Annotating type arguments and parameters
 
-You can annotate type arguments of generic types to provide nullability information for them as well.
+You can annotate type arguments and parameters of generic types to provide nullability information for them as well.
 For example, consider these annotations on a Java declaration:
 
 ```java
 @NotNull
-Set<@NotNull String> toSet(@NotNull Collection<@NotNull String> elements) { ... }
+<T> Collection<@NotNull String> collectNames(@NotNull Collection<@NotNull T> elements) { ... }
 ```
 
 It leads to the following signature seen in Kotlin:
 
 ```kotlin
-fun toSet(elements: (Mutable)Collection<String>) : (Mutable)Set<String> { ... }
+fun <T> collectNames(elements: (Mutable)Collection<T>) : (Mutable)Collection<String> { ... }
```

Comment:
Return type isn't exact `(Mutable)Collection<String>`. We don't have not null annotation on Java method's return type itself so in Kotlin it should be `(Mutable)Collection<String>!` (flexible).

Reply:
Moreover, ideally it should be:
```
fun <T> collectNames(elements: (Mutable)Collection<T!!>) : (Mutable)Collection<String>! { ... }
```
`T!!` means defenetely not null T (because we have the corresponding annotation on the Java side), but for now it isn't so. But with the special compiler flag `-XXLanguge:+ProhibitUsingNullableTypeParameterAgainstNotNullAnnotated` the behaviour turns into it.

Reply:
I wrote the example and checked the return value type in IDEA with and without specifying `@NotNull`:
- with `@NotNull` annotation on return type, I get `(Mutable)Collection<String>`
- with `@NotNull` annotation on return type, I get platform type `(Mutable)Collection<String>!`

🤔